rero_ils/modules/templates/permissions.py

```diff
@@ -0,0 +1,125 @@
+from rero_ils.modules.patrons.api import current_patron
+from rero_ils.modules.permissions import RecordPermission
+
+
+class TemplatePermission(RecordPermission):
+    """Template permissions."""
+
+    @classmethod
+    def list(cls, user, record=None):
+        """List permission check.
+
+        :param user: Logged user.
+        :param record: Record to check.
+        :return: True is action can be done.
+        """
+        # Operation allowed only for staff members (lib, sys_lib)
+        return current_patron and current_patron.is_librarian
+
+    @classmethod
+    def read(cls, user, record):
+        """Read permission check.
+
+        :param user: Logged user.
+        :param record: Record to check.
+        :return: True is action can be done.
+        """
+        # Check the user is authenticated and a record exists as param.
+        if not record or not current_patron:
+            return False
+        # only librarian or system_librarian can read templates of own org
+        if current_organisation['pid'] == record.organisation_pid:
+            #   - 'sys_librarian' could always read any record
+            if current_patron.is_system_librarian:
+                return True
+            #   - 'librarian' could only read his public and his own records.
+            elif current_patron.is_librarian:
+                if (
+                    record.is_public or
+                    (record.is_private and
+                        record.creator_pid == current_patron.pid)
+                ):
+                    return True
+        return False
+
+    @classmethod
+    def create(cls, user, record=None):
+        """Create permission check.
+
+        :param user: Logged user.
+        :param record: Record to check.
+        :return: True is action can be done.
+        """
+        # only librarian or system_librarian can create templates
+        if not current_patron or not current_patron.is_librarian:
+            return False
+        # ... only for its own organisation
+        if record:
+            if current_organisation['pid'] == record.organisation_pid:
+                if current_patron.is_system_librarian:
+                    return True
+                # librarian can only create private records
+                elif current_patron.is_librarian and record.is_private and \
+                        record.creator_pid == current_patron.pid:
+                    return True
+            return False
+        return True
+
+    @classmethod
+    def update(cls, user, record):
+        """Update permission check.
+
+        :param user: Logged user.
+        :param record: Record to check.
+        :return: True is action can be done.
+        """
+        # User must be be authenticated and have (at least) librarian role
+        if not current_patron or not current_patron.is_librarian:
+            return False
+        if current_patron and not record:  # legacy
+            return True
```

Comment:
After multiple test, I didn't find any issue about this legacy statement. As any other resources, for me, you can remove this statement. If you are agree with this fact, `create` and `update` methods are the same so you can only use one method inside other.

Reply:
I made the modif for the template resource only, we can do it for the other resources in another PR